x-pack/plugins/apm/public/components/shared/FilterBar/ServiceEnvFilter.tsx

```diff
@@ -0,0 +1,94 @@
+import { EuiFilterGroup, EuiFilterButton, EuiSelect } from '@elastic/eui';
+import React, { useEffect, useRef, useState } from 'react';
+import { useFetcher } from '../../../hooks/useFetcher';
+import { loadServiceEnvironments } from '../../../services/rest/apm/services';
+import { useLocation } from '../../../hooks/useLocation';
+import { useUrlParams } from '../../../hooks/useUrlParams';
+import { history } from '../../../utils/history';
+import { fromQuery, toQuery } from '../Links/url_helpers';
+
+function without<T>(obj: T, key: string): T {
+  const result: { [objKey: string]: unknown } = Object.assign({}, obj);
+  delete result[key];
+  return result as T;
+}
+
+export const ServiceEnvFilter: React.FC = () => {
+  const location = useLocation();
+  const { urlParams } = useUrlParams();
+  const { start, end, kuery, environment } = urlParams;
+  const { data = [] } = useFetcher(
+    () => loadServiceEnvironments({ start, end, kuery, environment: '' }),
```

Comment:
Not sure if we should load environment separately. Could we load it as part of the service details (like we do with `transaction.type`)?
Or should we make a dedicated filters endpoint?

Reply:
Actually we probably shouldn't bundle this to the service call. The filters will need to be fetched from ES separately.
But we could still have a single endpoint that fetches them. Perhaps:

```js
fetchFilters(filtersToFetch, uiFilters)
```
Where `filtersToFetch = ['kuery', 'environment', 'transactionType', 'host']` (or something like it).

Reply:
Actually, maybe loading `environment` separately for now is fine.

Reply:
we'll need to deal with the loading state for the selector component